### ISSUE TITLE
LIbrary cleaning / Fast Scanning

### ISSUE THREAD

**Issue by bossanova808** [matthuisman/docker-kodi-headless#28]
I have a scenario like this:

Server with bunch of NFS shares (10 of them)
Another server running dockers, including Kodi Headless dockers.  There are `sources.xml` files in place.

I have things working well in terms of library updates - i.e. new shows when dowloaded are inserted - but as my system will progressively download better quality copies, I am getting multiple inserts into my Kodi DB.  

If I run a clean inside the headless Kodi (i.e. a clean after update), the entire library disappears, because (I presume) the Kodi Docker can't see the network the NFS shares are on.  Thus I need to run a clean library on my Kodi clients whenever they starts up to get rid of the extra entries.  This is janky, obviously, and defeats some of the purpose of the headless Kodi instances.

I am wondering if you have any ideas on how I might go about solving this?  I could in theory mount the nfs shares into the docker, but I am not sure if the past substitution trick you offer in 'Fast Scanning' would then work??

(Or should I be able to run a clean, with external NFS sources?   To be honest it seems odd the inserts work, if the clean doesn't??)

**Comment by matthuisman** [comment]
How is it scanning new content? You passing it nfs:// paths? I suspect you just need to copy in a sources.xml from one of the clients

**Comment by matthuisman** [comment]
Sorry,  I see you mentioned sources.xml. hmmm

**Comment by bossanova808** [comment]
The new content is pushed from sonarr etc., and a scan is kicked off as I'd expect:

```
2023-01-19 16:49:54.216 T:398     DEBUG <CWebserver[8080]>: request received for /jsonrpc
2023-01-19 16:49:54.216 T:398     DEBUG <general>: connect replacing configured host mariadb-official with resolved host 172.18.0.5
2023-01-19 16:49:54.666 T:399     DEBUG <CWebserver[8080]>: Skipped 1 duplicate messages..
2023-01-19 16:49:54.666 T:399     DEBUG <CWebserver[8080]>: request received for /jsonrpc
2023-01-19 16:49:54.671 T:396     DEBUG <general>: connect replacing configured host mariadb-official with resolved host 172.18.0.5
2023-01-19 16:49:54.680 T:396     DEBUG <general>: Skipped 1 duplicate messages..
2023-01-19 16:49:54.680 T:396     DEBUG <general>: This query part contains a like, we will double backslash in the next field: ' AND idPath NOT IN (SELECT idPath FROM files WHERE strFileName LIKE 'video_ts.ifo') AND idPath NOT IN (SELECT idPath FROM files WHERE strFileName LIKE 'index.bdmv')
2023-01-19 16:49:54.726 T:396     DEBUG <general>: connect replacing configured host mariadb-official with resolved host 172.18.0.5
2023-01-19 16:49:54.730 T:396      INFO <general>: VideoInfoScanner: Starting scan ..
2023-01-19 16:49:55.631 T:396     DEBUG <general>: VideoInfoScanner: Rescanning dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/' due to change (cf7f490b050192e7a5137d23846d9278 != bef5923abd07b10e775ee1a1d8794d6b)
2023-01-19 16:49:55.633 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 01/Brother vs. Brother - S01E01 - Battle of the Bros WEBDL-1080p.mkv (s1e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.634 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 01/Brother vs. Brother - S01E02 - Weekend Warriors WEBDL-1080p.mkv (s1e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.634 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 01/Brother vs. Brother - S01E03 - Double Jeopardy WEBDL-1080p.mkv (s1e3) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.634 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 01/Brother vs. Brother - S01E04 - The Makeovers, Revealed WEBDL-1080p.mkv (s1e4) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.635 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 01/Brother vs. Brother - S01E05 - Flu Hits Home WEBDL-1080p.mkv (s1e5) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.636 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 01/Brother vs. Brother - S01E06 - Double Twin Powers WEBDL-1080p.mkv (s1e6) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.636 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 02/Brother vs. Brother - S02E01 - Season 2 Ranch Rematch WEBDL-1080p.mkv (s2e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.637 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 02/Brother vs. Brother - S02E02 - Cottage Shenanigans WEBDL-1080p.mkv (s2e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.638 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 02/Brother vs. Brother - S02E03 - Craftsman Double Trouble WEBDL-1080p.mkv (s2e3) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.639 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 02/Brother vs. Brother - S02E04 - Modern Colonials WEBDL-1080p.mkv (s2e4) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.639 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 02/Brother vs. Brother - S02E05 - Battle of the Ranchers WEBDL-1080p.mkv (s2e5) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.640 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 02/Brother vs. Brother - S02E06 - Tract Homes Showdown WEBDL-1080p.mkv (s2e6) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.641 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 03/Brother vs. Brother - S03E01 - Living Room Flip in Vegas HDTV-720p.mkv (s3e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.642 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 03/Brother vs. Brother - S03E02 - Kitchen Flip in Las Vegas HDTV-720p.mkv (s3e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.642 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 03/Brother vs. Brother - S03E03 - Bed and Bath Challenge HDTV-720p.mkv (s3e3) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.643 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 03/Brother vs. Brother - S03E04 - Las Vegas Flip Finale HDTV-720p.mkv (s3e4) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.644 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 04/Brother vs. Brother - S04E01 - Nice, Nice Baby SDTV.mkv (s4e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.644 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 04/Brother vs. Brother - S04E02 - If You Can't Take the Heat WEBDL-480p.mp4 (s4e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.645 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 04/Brother vs. Brother - S04E03 - Twin Telepathy WEBDL-480p.mp4 (s4e3) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.645 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 04/Brother vs. Brother - S04E04 - Family Affair WEBDL-480p.mp4 (s4e4) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.646 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 04/Brother vs. Brother - S04E05 - Makeover Magic WEBDL-480p.mp4 (s4e5) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.646 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 04/Brother vs. Brother - S04E06 - The Finale WEBDL-480p.mp4 (s4e6) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.647 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 05/Brother vs. Brother - S05E01 - Beachfront Battle WEBDL-720p.mkv (s5e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.648 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 05/Brother vs. Brother - S05E02 - Coastal Kitchens WEBRip-720p.mkv (s5e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.648 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 05/Brother vs. Brother - S05E03 - Seaside Masters WEBDL-720p.mkv (s5e3) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.649 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 05/Brother vs. Brother - S05E04 - Party-Ready Decks WEBRip-720p.mkv (s5e4) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.649 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 05/Brother vs. Brother - S05E05 - Sweet Guest Retreats WEBDL-720p.mkv (s5e5) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.649 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 05/Brother vs. Brother - S05E06 - The Finale WEBDL-720p.mkv (s5e6) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.649 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 06/Brother vs. Brother - S06E01 - San Francisco Bay Showdown WEBDL-1080p.mkv (s6e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.650 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 06/Brother vs. Brother - S06E02 - California Kitchens WEBDL-1080p.mkv (s6e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.651 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 06/Brother vs. Brother - S06E03 - Battle of the Master Suites WEBDL-1080p.mkv (s6e3) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.651 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 06/Brother vs. Brother - S06E04 - Slam Dunk Bonus Spaces WEBDL-1080p.mkv (s6e4) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.652 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 06/Brother vs. Brother - S06E05 - Guest Rooms for the Win WEBDL-1080p.mkv (s6e5) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.652 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 06/Brother vs. Brother - S06E06 - The Finale WEBDL-1080p.mkv (s6e6) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.653 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 07/Brother vs. Brother - S07E01 - Hollywood, Here We Come! WEBRip-720p.mkv (s7e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.654 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 07/Brother vs. Brother - S07E02 - A Tale of Two Kitchens WEBRip-720p.mkv (s7e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.654 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 07/Brother vs. Brother - S07E03 - Super Bonus Spaces WEBRip-720p.mkv (s7e3) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.655 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 07/Brother vs. Brother - S07E04 - Brother Bedroom Battle WEBRip-720p.mkv (s7e4) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.655 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 07/Brother vs. Brother - S07E05 - The Mystery Judge WEBRip-720p.mkv (s7e5) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.656 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 07/Brother vs. Brother - S07E06 - And the Winner Is WEBRip-720p.mkv (s7e6) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.656 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 08/Brother vs. Brother - S08E01 - No Rules! - WEBDL-1080p - h264 AAC 2.0.mkv (s8e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.657 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 08/Brother vs. Brother - S08E02 - Who's King of the Kitchens - WEBDL-1080p - h264 AAC 2.0.mkv (s8e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:55.782 T:396     DEBUG <general>: GetArtwork: Reading artwork for '271857' using The TVDB (new) scraper (file: '/config/.kodi/addons/metadata.tvdb.com.python/', content: 'tvshows', version: '1.0.3')
2023-01-19 16:49:55.783 T:396     DEBUG <general>: StartScript - calling plugin The TVDB (new)('plugin://metadata.tvdb.com.python/','5','?action=getartwork&id=271857&pathSettings=%7b%22RatingS%22%3a0%2c%22absolutenumber%22%3afalse%2c%22dvdorder%22%3afalse%2c%22language%22%3a%22en%22%2c%22token%22%3a%22eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MTYyODI3NTEsImlkIjoiS29kaSIsIm9yaWdfaWF0IjoxNjE1Njc3OTUxfQ.zqeQ2JpaillB_skXnYb0_wq-ANule3wzpXrJij56MTtuJQbDUQI2qf3EHP5lW3ChWduEdh2dDF_gm_ZD4v6AWPxT1IGnoFWTCyhECH6jOCOxKKfWKNllCO0cH38uHR22F-isBa44OmwPGIifqNh2CYVi6kwZIQbG6JNQSkc_Zyx0q5O7iCxjAwQMGtgBeJpffVWLpR1Z9qI-T6auMxorONCMu1FY16JySYxqiovxML1FBqw8HsQPjiCdALY486kMlXfay5F6Nzc9qLvlZQh2gATc4wqluTyZYwPAoeYYFiN0CyG60PJbNWXbL58WLRletfx5sHcQsSuLJQ5yCVHm0A%22%7d','resume:false')
2023-01-19 16:49:55.787 T:396     DEBUG <general>: GetLanguageInvoker - Reusing LanguageInvokerThread 5 for script /config/.kodi/addons/metadata.tvdb.com.python/scraper.py
2023-01-19 16:49:55.788 T:397      INFO <general>: initializing python engine.
2023-01-19 16:49:55.788 T:397     DEBUG <general>: CPythonInvoker(5, /config/.kodi/addons/metadata.tvdb.com.python/scraper.py): start processing
2023-01-19 16:49:55.788 T:397     DEBUG <general>: CPythonInvoker(5, /config/.kodi/addons/metadata.tvdb.com.python/scraper.py): entering source directory /config/.kodi/addons/metadata.tvdb.com.python
2023-01-19 16:49:55.788 T:397     DEBUG <general>: CPythonInvoker(5, /config/.kodi/addons/metadata.tvdb.com.python/scraper.py): instantiating addon using automatically obtained id of "metadata.tvdb.com.python" dependent on version 3.0.0 of the xbmc.python api
2023-01-19 16:49:55.789 T:397     DEBUG <general>: metadata.tvdb.com.python: Called addon with params: ['plugin://metadata.tvdb.com.python/', '5', '?action=getartwork&id=271857&pathSettings=%7b%22RatingS%22%3a0%2c%22absolutenumber%22%3afalse%2c%22dvdorder%22%3afalse%2c%22language%22%3a%22en%22%2c%22token%22%3a%22eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MTYyODI3NTEsImlkIjoiS29kaSIsIm9yaWdfaWF0IjoxNjE1Njc3OTUxfQ.zqeQ2JpaillB_skXnYb0_wq-ANule3wzpXrJij56MTtuJQbDUQI2qf3EHP5lW3ChWduEdh2dDF_gm_ZD4v6AWPxT1IGnoFWTCyhECH6jOCOxKKfWKNllCO0cH38uHR22F-isBa44OmwPGIifqNh2CYVi6kwZIQbG6JNQSkc_Zyx0q5O7iCxjAwQMGtgBeJpffVWLpR1Z9qI-T6auMxorONCMu1FY16JySYxqiovxML1FBqw8HsQPjiCdALY486kMlXfay5F6Nzc9qLvlZQh2gATc4wqluTyZYwPAoeYYFiN0CyG60PJbNWXbL58WLRletfx5sHcQsSuLJQ5yCVHm0A%22%7d', 'resume:false']
2023-01-19 16:49:56.945 T:397   WARNING <general>: Attempt to use invalid handle 5
2023-01-19 16:49:56.945 T:397      INFO <general>: CPythonInvoker(5, /config/.kodi/addons/metadata.tvdb.com.python/scraper.py): script successfully run
2023-01-19 16:49:56.945 T:396     ERROR <general>: GetDirectory - Error getting 
2023-01-19 16:49:56.959 T:396     DEBUG <general>: Mysql execute: update path set strHash='bef5923abd07b10e775ee1a1d8794d6b' where idPath=923
2023-01-19 16:49:56.989 T:396     DEBUG <general>: VideoInfoScanner: Skipping dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 01/' due to no change
2023-01-19 16:49:57.015 T:396     DEBUG <general>: VideoInfoScanner: Skipping dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 02/' due to no change
2023-01-19 16:49:57.037 T:396     DEBUG <general>: VideoInfoScanner: Skipping dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 03/' due to no change
2023-01-19 16:49:57.062 T:396     DEBUG <general>: VideoInfoScanner: Skipping dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 04/' due to no change
2023-01-19 16:49:57.088 T:396     DEBUG <general>: VideoInfoScanner: Skipping dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 05/' due to no change
2023-01-19 16:49:57.115 T:396     DEBUG <general>: VideoInfoScanner: Skipping dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 06/' due to no change
2023-01-19 16:49:57.142 T:396     DEBUG <general>: VideoInfoScanner: Skipping dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 07/' due to no change
2023-01-19 16:49:57.171 T:396     DEBUG <general>: VideoInfoScanner: Rescanning dir 'nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 08/' due to change (255de7cd6710367abfef98d6f1eda3d0 != a8240818502168d4fedf9c75d9e19fea)
2023-01-19 16:49:57.172 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 08/Brother vs. Brother - S08E01 - No Rules! - WEBDL-1080p - h264 AAC 2.0.mkv (s8e1) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:57.173 T:396     DEBUG <general>: VideoInfoScanner: Found episode match nfs://192.168.1.51/TVLibrary05/Brother vs. Brother/Season 08/Brother vs. Brother - S08E02 - Who's King of the Kitchens - WEBDL-1080p - h264 AAC 2.0.mkv (s8e2) [s([0-9]+)[ ._x-]*e([0-9]+(?:(?:[a-i]|\.[1-9])(?![0-9]))?)([^\\/]*)$]
2023-01-19 16:49:57.206 T:396     DEBUG <general>: GetArtwork: Reading artwork for '271857' using The TVDB (new) scraper (file: '/config/.kodi/addons/metadata.tvdb.com.python/', content: 'tvshows', version: '1.0.3')
2023-01-19 16:49:57.206 T:396     DEBUG <general>: StartScript - calling plugin The TVDB (new)('plugin://metadata.tvdb.com.python/','5','?action=getartwork&id=271857&pathSettings=%7b%22RatingS%22%3a0%2c%22absolutenumber%22%3afalse%2c%22dvdorder%22%3afalse%2c%22language%22%3a%22en%22%2c%22token%22%3a%22eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MTYyODI3NTEsImlkIjoiS29kaSIsIm9yaWdfaWF0IjoxNjE1Njc3OTUxfQ.zqeQ2JpaillB_skXnYb0_wq-ANule3wzpXrJij56MTtuJQbDUQI2qf3EHP5lW3ChWduEdh2dDF_gm_ZD4v6AWPxT1IGnoFWTCyhECH6jOCOxKKfWKNllCO0cH38uHR22F-isBa44OmwPGIifqNh2CYVi6kwZIQbG6JNQSkc_Zyx0q5O7iCxjAwQMGtgBeJpffVWLpR1Z9qI-T6auMxorONCMu1FY16JySYxqiovxML1FBqw8HsQPjiCdALY486kMlXfay5F6Nzc9qLvlZQh2gATc4wqluTyZYwPAoeYYFiN0CyG60PJbNWXbL58WLRletfx5sHcQsSuLJQ5yCVHm0A%22%7d','resume:false')
2023-01-19 16:49:57.207 T:396     DEBUG <general>: GetLanguageInvoker - Reusing LanguageInvokerThread 5 for script /config/.kodi/addons/metadata.tvdb.com.python/scraper.py
2023-01-19 16:49:57.207 T:397      INFO <general>: initializing python engine.
2023-01-19 16:49:57.207 T:397     DEBUG <general>: CPythonInvoker(5, /config/.kodi/addons/metadata.tvdb.com.python/scraper.py): start processing
2023-01-19 16:49:57.208 T:397     DEBUG <general>: CPythonInvoker(5, /config/.kodi/addons/metadata.tvdb.com.python/scraper.py): entering source directory /config/.kodi/addons/metadata.tvdb.com.python
2023-01-19 16:49:57.208 T:397     DEBUG <general>: CPythonInvoker(5, /config/.kodi/addons/metadata.tvdb.com.python/scraper.py): instantiating addon using automatically obtained id of "metadata.tvdb.com.python" dependent on version 3.0.0 of the xbmc.python api
2023-01-19 16:49:57.208 T:397     DEBUG <general>: metadata.tvdb.com.python: Called addon with params: ['plugin://metadata.tvdb.com.python/', '5', '?action=getartwork&id=271857&pathSettings=%7b%22RatingS%22%3a0%2c%22absolutenumber%22%3afalse%2c%22dvdorder%22%3afalse%2c%22language%22%3a%22en%22%2c%22token%22%3a%22eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2MTYyODI3NTEsImlkIjoiS29kaSIsIm9yaWdfaWF0IjoxNjE1Njc3OTUxfQ.zqeQ2JpaillB_skXnYb0_wq-ANule3wzpXrJij56MTtuJQbDUQI2qf3EHP5lW3ChWduEdh2dDF_gm_ZD4v6AWPxT1IGnoFWTCyhECH6jOCOxKKfWKNllCO0cH38uHR22F-isBa44OmwPGIifqNh2CYVi6kwZIQbG6JNQSkc_Zyx0q5O7iCxjAwQMGtgBeJpffVWLpR1Z9qI-T6auMxorONCMu1FY16JySYxqiovxML1FBqw8HsQPjiCdALY486kMlXfay5F6Nzc9qLvlZQh2gATc4wqluTyZYwPAoeYYFiN0CyG60PJbNWXbL58WLRletfx5sHcQsSuLJQ5yCVHm0A%22%7d', 'resume:false']
2023-01-19 16:49:58.207 T:397      INFO <general>: CPythonInvoker(5, /config/.kodi/addons/metadata.tvdb.com.python/scraper.py): script successfully run
2023-01-19 16:49:58.207 T:396     ERROR <general>: GetDirectory - Error getting 
2023-01-19 16:49:58.224 T:396     DEBUG <general>: Mysql execute: update path set strHash='a8240818502168d4fedf9c75d9e19fea' where idPath=3694
2023-01-19 16:49:58.228 T:396      INFO <general>: VideoInfoScanner: Finished scan. Scanning for video info took 00:03
2023-01-19 16:50:28.233 T:396     DEBUG <general>: Thread JobWorker 139854031623936 terminating (autodelete)
2023-01-19 16:51:28.738 T:327      INFO <general>: NFS is idle. Closing the remaining connections.
```

**Comment by matthuisman** [comment]
what does the clean log say?
you do have the sources.xml correctly installed on headless?

as the scraper setup info actually comes from the mysql db (not sources)

**Comment by bossanova808** [comment]
I'll get a clean log in a day or two, just working on a few nexus things as I shift the house over.

`sources.xml` is in the correct place, i.e. alongside `advancedsettings.xml`

**Comment by bossanova808** [comment]
Hmm, I think I worked it out.  I had a path substitution in `adavancedsettings.xml` for `sources.xml` it seems (I use as much centralised config as I can as it's very tedious keeping multiple installs working across years with Kodi....great program, but also egregiously fiddly in many ways!).  I probably should have been more careful with my cut and paste when I set up these headless instances.  I did also have the actual file, but...

Anyhoo, it appears this path sub was not working with the headess instances.  I guess I must have assumed it was (or simply not noticed this) - as the inserts were working, but now realise that comes from the scrapers.  Once I saw I was doing this, I removed the path sub and tried with an _actual_ `sources.xml` in place and turned on `cleanafterupdate` and....it appears to have worked ok.

So I think I might be ok here...I'll close this, will pop back if in fact it doesn't keep working, but I think it was PEBKAC really.
